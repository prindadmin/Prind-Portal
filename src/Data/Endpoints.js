export const DEFAULTPAGE = ""
export const DEFAULTLOGGEDINPAGE = "/inception"
export const PROJECTDETAILSPAGE = "/project"
export const SIGNINPAGE = "/signin"
export const SIGNUPPAGE = "/signup"
export const MEMBERPAGE = "/team"
