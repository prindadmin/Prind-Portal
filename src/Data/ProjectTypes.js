import * as Strings from './Strings'

const ProjectTypesList = [
  {
    id: 'CDM2015Project',
    icon: 'building',
    name: Strings.PROJECT_TYPE_CDM2015PROJECT_NAME,
    description: Strings.PROJECT_TYPE_CDM2015PROJECT_DESCRIPTION
  },
  {
    id: 'DHSFProject',
    icon: 'building',
    name: Strings.PROJECT_TYPE_DHSFPROJECT_NAME,
    description: Strings.PROJECT_TYPE_DHSFPROJECT_DESCRIPTION
  }
]

export default ProjectTypesList
