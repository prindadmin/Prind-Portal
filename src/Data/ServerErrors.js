
export const PROJECT_NOT_FOUND = "PROJECT_NOT_FOUND"
export const INSUFFICIENT_PERMISSION = "INSUFFICIENT_PERMISSION"
