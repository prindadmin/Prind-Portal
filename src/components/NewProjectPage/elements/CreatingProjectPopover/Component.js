import React, { Component } from 'react'
//import PropTypes from 'prop-types'

import PopOverHandler from '../../../common/popOverHandler'

import {
  Spinner,
  Intent,
} from '@blueprintjs/core'

import * as strings from '../../../../data/Strings'

export class Element extends Component {
  static propTypes = {
  }

  getContent = () => {
    return (
      <React.Fragment>
        {strings.PROJECT_CREATION_BILLING_WARNING}
        <Spinner size='150' intent={Intent.DANGER}/>
      </React.Fragment>
    )
  }

  render () {
    return (
      <PopOverHandler>
        <div id='popup-greyer' onClick={(e) => e.stopPropagation()}>
          <div id='project-creation-popover'>
            <div id='popup-box'>
              <div className='project-creation-popover-container' onClick={(e) => e.stopPropagation()}>
                <div className='element-title'>
                  {strings.PROJECT_CREATION_IN_PROGRESS}
                </div>
                <div className='element-description'>
                  {this.getContent()}
                </div>
              </div>
            </div>
          </div>
        </div>
      </PopOverHandler>
    )
  }

}

export default Element
