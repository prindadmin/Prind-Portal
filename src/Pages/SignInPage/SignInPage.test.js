import React from 'react';
import { shallow } from 'enzyme';
import ReactGA from 'react-ga';

import Component from './SignInPage'
import * as Endpoints from '../../Data/Endpoints'

beforeAll(() => {
  ReactGA.initialize('dummy', { testMode: true });
});

it('Sign In Page - Should render sign in page', () => {
  const props = {
    location: {
      pathname: Endpoints.SIGNINPAGE,
      state: {}
    }
  };
  const component = shallow(<Component {...props} />);
  expect(component).toMatchSnapshot();
});

it('Sign In Page - Should render sign in page without state', () => {
  const props = {
    location: {
      pathname: Endpoints.SIGNINPAGE,
    }
  };
  const component = shallow(<Component {...props} />);
  expect(component).toMatchSnapshot();
});

it('Sign In Page - Should render sign in page without state then add state', () => {
  const props = {
    location: {
      pathname: Endpoints.SIGNINPAGE,
    }
  };
  const component = shallow(<Component {...props} />);
  expect(component).toMatchSnapshot();
  component.setProps({
    location: {
      pathname: Endpoints.SIGNINPAGE,
      state: {
        username: "aaaa-aaaa-aaaaaaaa-aaaa-aaaa"
      }
    }
  })
  expect(component).toMatchSnapshot();
});

it('Sign In Page - Should render sign in page without state then change page', () => {
  const props = {
    location: {
      pathname: Endpoints.SIGNINPAGE,
    }
  };
  const component = shallow(<Component {...props} />);
  expect(component).toMatchSnapshot();
  component.setProps({
    location: {
      pathname: Endpoints.SIGNUPPAGE,
    }
  })
  expect(component).toMatchSnapshot();
});

/*
// TODO: LOW PRIORITY: Test Can use webp
it('Sign In Page - Should render with webp', () => {
  const webP = jest.mock('../../Components/../Functions/CheckIfWebpSupported')

  webP.default.mockResolvedValue(true)

  const props = {
    location: {
      pathname: Endpoints.SIGNINPAGE,
    }
  };
  const component = shallow(<Component {...props} />);
  expect(component).toMatchSnapshot();
});
*/
