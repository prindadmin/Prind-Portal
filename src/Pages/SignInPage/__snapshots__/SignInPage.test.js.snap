// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign In Page - Should render sign in page 1`] = `
<div
  className="full-width full-height"
  id="sign-in-up-page"
  style={
    Object {
      "backgroundImage": "url(/images/backgrounds/building-site-1.png)",
    }
  }
>
  <withRouter(Connect(SignInUpBox))
    backgroundColor="#FFFFFF"
    isForgotPassword={false}
    isPasswordReset={false}
    isSignIn={true}
    isSignUp={false}
    reference="sign-in-up-box"
  />
</div>
`;

exports[`Sign In Page - Should render sign in page without state 1`] = `
<div
  className="full-width full-height"
  id="sign-in-up-page"
  style={
    Object {
      "backgroundImage": "url(/images/backgrounds/building-site-1.png)",
    }
  }
>
  <withRouter(Connect(SignInUpBox))
    backgroundColor="#FFFFFF"
    isForgotPassword={false}
    isPasswordReset={false}
    isSignIn={true}
    isSignUp={false}
    reference="sign-in-up-box"
  />
</div>
`;

exports[`Sign In Page - Should render sign in page without state then add state 1`] = `
<div
  className="full-width full-height"
  id="sign-in-up-page"
  style={
    Object {
      "backgroundImage": "url(/images/backgrounds/building-site-1.png)",
    }
  }
>
  <withRouter(Connect(SignInUpBox))
    backgroundColor="#FFFFFF"
    isForgotPassword={false}
    isPasswordReset={false}
    isSignIn={true}
    isSignUp={false}
    reference="sign-in-up-box"
  />
</div>
`;

exports[`Sign In Page - Should render sign in page without state then add state 2`] = `
<div
  className="full-width full-height"
  id="sign-in-up-page"
  style={
    Object {
      "backgroundImage": "url(/images/backgrounds/building-site-1.png)",
    }
  }
>
  <withRouter(Connect(SignInUpBox))
    backgroundColor="#FFFFFF"
    isForgotPassword={false}
    isPasswordReset={false}
    isSignIn={true}
    isSignUp={false}
    reference="sign-in-up-box"
    username="aaaa-aaaa-aaaaaaaa-aaaa-aaaa"
  />
</div>
`;

exports[`Sign In Page - Should render sign in page without state then change page 1`] = `
<div
  className="full-width full-height"
  id="sign-in-up-page"
  style={
    Object {
      "backgroundImage": "url(/images/backgrounds/building-site-1.png)",
    }
  }
>
  <withRouter(Connect(SignInUpBox))
    backgroundColor="#FFFFFF"
    isForgotPassword={false}
    isPasswordReset={false}
    isSignIn={true}
    isSignUp={false}
    reference="sign-in-up-box"
  />
</div>
`;

exports[`Sign In Page - Should render sign in page without state then change page 2`] = `
<div
  className="full-width full-height"
  id="sign-in-up-page"
  style={
    Object {
      "backgroundImage": "url(/images/backgrounds/building-site-1.png)",
    }
  }
>
  <withRouter(Connect(SignInUpBox))
    backgroundColor="#FFFFFF"
    isForgotPassword={false}
    isPasswordReset={false}
    isSignIn={false}
    isSignUp={true}
    reference="sign-in-up-box"
  />
</div>
`;

exports[`Sign In Page - Should render with webp 1`] = `
<div
  className="full-width full-height"
  id="sign-in-up-page"
  style={
    Object {
      "backgroundImage": "url(/images/backgrounds/building-site-1.webp)",
    }
  }
>
  <withRouter(Connect(SignInUpBox))
    backgroundColor="#FFFFFF"
    isForgotPassword={false}
    isPasswordReset={false}
    isSignIn={true}
    isSignUp={false}
    reference="sign-in-up-box"
  />
</div>
`;
